cli/src/pcluster/validators/cluster_validators.py

```diff
@@ -1336,6 +1336,24 @@ def _validate(self, head_node_instance_type: str, total_max_compute_nodes: int):
             )
 
 
+class SharedStorageEfsSettingsValidator(Validator):
+    """
+    HeadNode SharedStorageEfsSettings Validator.
+
+    Verify HeadNode SharedStorageEfsSettings can only be used with Efs SharedStorageType.
+    """
+
+    def _validate(self, shared_storage_type: str, shared_storage_efs_settings):
+        if shared_storage_efs_settings and shared_storage_type.lower() != SharedStorageType.EFS.value:
+            self._add_failure(
+                "SharedStorageEfsSettings is specified "
+                f"but the SharedStorageType is set to {shared_storage_type}. "
+                "SharedStorageEfsSettings can only be used when SharedStorageType "
+                f"is specified as Efs.",
```

Comment:
LEt's use the enum rather than rewriting 'Efs'. Remember that you can capitalize the string.
Also, we can make it shorter and say 'SharedStoragerType is Efs' ratherr than 'is specified as'